presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorColumnIdentity.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor;
+
+import com.facebook.presto.spi.ColumnIdentity;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.primitives.Longs;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.io.BaseEncoding.base16;
+import static com.google.common.primitives.Longs.fromByteArray;
+import static java.lang.String.format;
+import static java.lang.System.arraycopy;
+import static java.util.Arrays.copyOfRange;
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+public class RaptorColumnIdentity
+        implements ColumnIdentity
+{
+    private static final byte CURRENT_VERSION = 1;
+
+    private final byte version;
+    private final long columnId;
+
+    public RaptorColumnIdentity(long columnId)
+    {
+        this(CURRENT_VERSION, columnId);
+    }
+
+    private RaptorColumnIdentity(byte version, long columnId)
+    {
+        checkArgument(version > 0 && version <= CURRENT_VERSION, "version must be a positive integer less than or equal to %s", CURRENT_VERSION);
```

Comment:
Just say `"invalid column identity version: %s"`

(This is caused by a bug or corrupt data: the user isn't going to go and change the value based on the error message)